frontend/e2e-test/test/e2e/specs/2_employee-2/application-transitions.spec.ts

```diff
@@ -295,3 +296,32 @@ test('Placement proposal flow', async (t) => {
   await unitPage.openTabWaitingConfirmation()
   await t.expect(unitPage.waitingGuardianConfirmationRow.count).eql(1)
 })
+
+test('Supervisor cannot download decision PDF while it is being generated', async (t) => {
+  const fixture = {
+    ...applicationFixture(
+      fixtures.enduserChildFixtureJari,
+      fixtures.enduserGuardianFixture
+    ),
+    status: 'SENT' as const
+  }
+  applicationId = fixture.id
+
+  await insertApplications([fixture])
+
+  await t.useRole(seppoManagerRole)
+
+  // NOTE: NOT running pending async jobs yet -> PDF won't have been generated from sent decision
+  await execSimpleApplicationActions(applicationId, [
+    'move-to-waiting-placement',
+    'create-default-placement-plan',
+    'send-decisions-without-proposal'
+  ])
+
+  await applicationReadView.openApplicationByLink(applicationId)
+  await applicationReadView.assertDecisionDownloadPending('DAYCARE')
+
+  await runPendingAsyncJobs()
```

Comment:
I wonder if this will sometimes fail because the async job would happen to finish too soon. If I'm correct, `runPendingAsyncJobs` ensures that pending jobs have been run, but they _may_ have been run before that already anyway.

Reply:
Ah, that's true. I'll check if there's some way around this.

Reply:
If it doesn't seem to be failing in practice you could also just leave a comment so if it later becomes unstable the reason is easy to figure out

Reply:
Decided to rewrite this test with different (and new) Dev API endpoints which should now properly control when the PDF generation is triggered, making this stable.